hack/lib/golang.sh

```diff
@@ -443,9 +444,17 @@ kube::golang::build_binaries_for_platform() {
     fi
   fi
 
+  teststale=$(kube::golang::output_filename_for_binary "hack/cmd/teststale" "${platform}")
   for test in "${tests[@]:+${tests[@]}}"; do
     local outfile=$(kube::golang::output_filename_for_binary "${test}" \
       "${platform}")
+    local testpkg="$(dirname ${test})"
+    if [[ "$(${teststale} -binary "${outfile}" -package "${testpkg}")" == "false" ]]; then
+      continue
+    fi
+    go install "${goflags[@]:+${goflags[@]}}" \
```

Comment:
Thanks for pointing this out. This definitely deserves a comment. I spent half a day trying to figure this out. Writing everything else took much less time. :)

`go test -c` below directly builds the binary. It builds the packages, but it never installs them. `go test -i` only installs the dependencies of the test, but not the test package itself. So neither `go test -c` nor `go test -i` install, for example, test/e2e.a. And without that, doing a staleness check on `k8s.io/kubernetes/test/e2e` package always returns true (always stale). That's why we need to install that package.

I will add that as a comment.